wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -58,8 +58,15 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
       res <- resF
 
       hash = block.blockHeader.hashBE
-      height <- chainQueryApi.getBlockHeight(hash)
-      _ <- stateDescriptorDAO.updateSyncHeight(hash, height.get)
+      heightOpt <- chainQueryApi.getBlockHeight(hash)
+      _ <- heightOpt match {
+        case Some(height) =>
+          stateDescriptorDAO.updateSyncHeight(hash, height)
+        case None =>
+          logger.warn(
```

Comment:
probably should upgrade this as an `error` log as there really is a big problem if callbacks are being executed on blocks we don't have.